src/rabbit_control_main.erl

```diff
@@ -499,62 +499,71 @@ action(purge_queue, Node, [Q], Opts, Inform, Timeout) ->
 
 action(list_users, Node, [], _Opts, Inform, Timeout) ->
     Inform("Listing users", []),
-    display_info_list(
-      call(Node, {rabbit_auth_backend_internal, list_users, []}, Timeout),
-      rabbit_auth_backend_internal:user_info_keys());
+    call(Node, {rabbit_auth_backend_internal, list_users, []}, Ref=make_ref(),
+         Timeout),
+    wait_for_info_messages(self(), Ref,
+                           rabbit_auth_backend_internal:user_info_keys(),
+                           Timeout);
 
 action(list_permissions, Node, [], Opts, Inform, Timeout) ->
     VHost = proplists:get_value(?VHOST_OPT, Opts),
     Inform("Listing permissions in vhost \"~s\"", [VHost]),
-    display_info_list(call(Node, {rabbit_auth_backend_internal,
-                             list_vhost_permissions, [VHost]}, Timeout),
-                      rabbit_auth_backend_internal:vhost_perms_info_keys());
+    call(Node,{rabbit_auth_backend_internal, list_vhost_permissions,
+               [VHost]}, Ref=make_ref(), Timeout),
+    wait_for_info_messages(self(), Ref,
+                           rabbit_auth_backend_internal:vhost_perms_info_keys(),
+                           Timeout);
 
 action(list_parameters, Node, [], Opts, Inform, Timeout) ->
     VHostArg = list_to_binary(proplists:get_value(?VHOST_OPT, Opts)),
     Inform("Listing runtime parameters", []),
-    display_info_list(
-      rpc_call(Node, rabbit_runtime_parameters, list_formatted, [VHostArg],
-               Timeout),
-      rabbit_runtime_parameters:info_keys());
+    spawn_link(rabbit_cli, rpc_call, [Node, rabbit_runtime_parameters,
+                                      list_formatted, [VHostArg], Ref=make_ref(),
+                                      Pid=self(), Timeout]),
+    wait_for_info_messages(Pid, Ref, rabbit_runtime_parameters:info_keys(), Timeout);
 
 action(list_policies, Node, [], Opts, Inform, Timeout) ->
     VHostArg = list_to_binary(proplists:get_value(?VHOST_OPT, Opts)),
     Inform("Listing policies", []),
-    display_info_list(rpc_call(Node, rabbit_policy, list_formatted, [VHostArg],
-                              Timeout),
-                      rabbit_policy:info_keys());
+    spawn_link(rabbit_cli, rpc_call, [Node, rabbit_policy, list_formatted,
```

Comment:
is there a reason why we have all these spawn_link calls, and also the new `call/4` down below? Is `call/4` meant to be used everywhere? 

Also there's seems to be a pattern here of `spawn_link` then `wait_for_info_messages`. Perhaps this can be refactored?